xonsh/lexer.py

```diff
@@ -33,53 +327,26 @@ def __init__(self, errfunc=lambda e, l, c: print(e)):
         self.errfunc = errfunc
         self.fname = ''
         self.last = None
-        self.lexer = None
-        self.indent = ''
-        self.in_py_mode = [True]
 
     def build(self, **kwargs):
         """Part of the PLY lexer API."""
-        self.lexer = lex.lex(object=self, **kwargs)
-        self.reset()
+        pass
 
     def reset(self):
-        self.lexer.lineno = 1
-        self.indent = ''
-        self.last = None
-        self.in_py_mode = [True]
-        self.in_parens = [False]
-
-    @property
-    def lineno(self):
-        if self.lexer is not None:
-            return self.lexer.lineno
-
-    @lineno.setter
-    def lineno(self, value):
-        if self.lexer is not None:
-            self.lexer.lineno = value
+        pass
 
     def input(self, s):
         """Calls the lexer on the string s."""
-        self.lexer.input(s)
+        s = re.sub(r'#.*', '', s)
```

Comment:
I had a lot of trouble with just using tokens to get rid of comments.  In particular, something like `echo "hello # comment` broke pretty easily, because when wrapped to try in subprocess mode, it was interpreted as `$[echo "hello" # comment ]`, which killed the last bracket and made it fail.

This was a quick hack that seemed to get to the right answer; if there's a more elegant way, I'm all for it!